src/NuGet.Indexing/StorageLoader.cs

```diff
@@ -7,21 +7,34 @@
 using Newtonsoft.Json;
 using System;
 using System.IO;
+using System.Threading.Tasks;
 using FrameworkLogger = Microsoft.Extensions.Logging.ILogger;
 
 namespace NuGet.Indexing
 {
     public class StorageLoader : ILoader
     {
-        private readonly CloudStorageAccount _storageAccount;
         private readonly string _containerName;
         private readonly FrameworkLogger _logger;
 
-        public StorageLoader(CloudStorageAccount storageAccount, string containerName, FrameworkLogger logger)
+        private CloudStorageAccount _storageAccount;
+        private AzureDirectorySynchronizer _synchronizer;
+        private Func<Task<Tuple<CloudStorageAccount, AzureDirectorySynchronizer>>> _createCloud;
+
+        public static async Task<StorageLoader> CreateStorageLoader(Func<Task<Tuple<CloudStorageAccount, AzureDirectorySynchronizer>>> createCloud, string containerName, FrameworkLogger logger)
```

Comment:
I'm not sure I like passing in a function here. I feel like StorageLoader itself should have the implementation to generate what it needs, and we should be passing in parameters that it needs (or the configuration). Is there a particular reason this is implemented in this manner?